src/ashley/models.py

```diff
@@ -139,6 +140,11 @@ def sync_user_groups(self, user, roles: List[str]) -> None:
         """
         Synchronize the group membership of a user for this LTI Context
         """
+        # Control user is part of this lti_context
+        if user.lti_consumer_id != self.lti_consumer_id:
+            raise PermissionDenied(
+                _("The User and LTIContext must be part of the same LTI Consumer")
+            )
```

Comment:
```suggestion
            raise PermissionDenied("The User and LTIContext must be part of the same LTI Consumer")
```

You don't need to translate exceptions.